### PR TITLE
Only call make gl context current if not already current

### PR DESCRIPTION
**Pull request by iskakaushik**
Before: [go/gl-context-before-fltt](https://goto.google.com/gl-context-before-fltt) took ~2.6 seconds

After: [go/gl-context-after-fltt](https://goto.google.com/gl-context-after-fltt) took ~7 ms

### CODE REVIEW

**Comment by dnfield** [comment]
Change looks good but should have a test.

One way to test this woul dbe to delegate the eglMakeCurrent call and make sure it doesn't get called when eglGetCurrent returns the current one.

**Comment by dnfield** [comment]
I think at worst this should be freeing up CPU time where we are currently blocking on a buffer dequeue, but I'm curious how much raster time it actually gets us.

File: shell/platform/android/android_context_gl.cc [hunk 1]
```diff
@@ -253,6 +253,10 @@ bool AndroidEGLSurface::IsValid() const {
 }
 
 bool AndroidEGLSurface::MakeCurrent() const {
+  EGLContext current_egl_context = eglGetCurrentContext();
```
Comment (jason-simmons) [inline shell/platform/android/android_context_gl.cc:256]: Check `eglGetCurrentSurface` in addition to the EGL context

**Comment by iskakaushik** [comment]
@dnfield  I was looking into writing a test for this. This seems very close to the OpenGL + Android specific, there aren't any tests that test things in AndroidEGLSurface. If we were to write a test that ensures that we would not call MakeCurrent if context is current, we would want to essentially mock out `IsContextCurrent` altogether which seems to add very little value.  

There aren't any tests for `CreatePbufferSurface` which will be the real test for this situation. I'm open to other ideas on what a good place to test would be.

**Comment by dnfield** [comment]
I think you can add a test in `android_context_gl_unittests` that does something like this:

```cpp
GrMockOptions main_context_options;
  sk_sp<GrDirectContext> main_context =
      GrDirectContext::MakeMock(&main_context_options);
  auto environment = fml::MakeRefCounted<AndroidEnvironmentGL>();
  std::string thread_label =
      ::testing::UnitTest::GetInstance()->current_test_info()->name();

  ThreadHost thread_host(ThreadHost::ThreadHostConfig(
      thread_label,
      ThreadHost::Type::UI | ThreadHost::Type::RASTER | ThreadHost::Type::IO));
  TaskRunners task_runners = MakeTaskRunners(thread_label, thread_host);
  auto context = std::make_unique<AndroidContextGL>(
      AndroidRenderingAPI::kOpenGLES, environment, task_runners, 4);
  auto pbuffer = context->CreatePBufferSurface();
// etc.
```

**Review by dnfield — APPROVED** [review]

**Pull request merged**